Suvorov.LNU.TwitterClone.Core/DIConfiguration.cs

```diff
@@ -0,0 +1,29 @@
+﻿using AutoMapper;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Suvorov.LNU.TwitterClone.Core.Interfaces;
+using Suvorov.LNU.TwitterClone.Core.Mapper;
+using Suvorov.LNU.TwitterClone.Models.Configuration;
+
+namespace Suvorov.LNU.TwitterClone.Core
+{
+    public static class DIConfiguration
+{
+    public static void RegisterCoreDependencies(this IServiceCollection services)
+    {
+            services.AddSingleton<IMapperProvider, MapperProvider>();
+            services.AddSingleton(GetMapper);
+    }
```

Comment:
Inconsistent Naming Convention: The class name DIConfiguration follows PascalCase convention, which is generally used for class names, while the method names RegisterCoreDependencies and RegisterCoreConfiguration follow camelCase convention, which is commonly used for method names. It is recommended to maintain a consistent naming convention throughout the codebase for clarity and consistency.
Lack of Dependency Injection Best Practices: The RegisterCoreDependencies method registers the IMapperProvider as a singleton using AddSingleton method, which is generally appropriate. However, the GetMapper method is registered as a singleton using AddSingleton without specifying an instance or a factory method. This can cause issues since it expects an instance of IMapperProvider to be available immediately, even before the service provider is fully configured. It would be better to use a factory method or a lambda expression to instantiate and provide the IMapperProvider instance during the registration process.